tests/test_config_parsing/test_parser_error_collection.py

```diff
@@ -0,0 +1,221 @@
+    errors_matching_location = [
+        x
+        for x in errors_matching_filename
+        if equals_or_expected_any(x.line, expected_line)
+        and equals_or_expected_any(x.column, expected_column)
+        and equals_or_expected_any(x.end_column, expected_end_column)
+    ]
+
+    assert len(errors_matching_location) > 0, (
+        f"Expected to find at least 1 error matching location"
+        f"(line={'*' if expected_line is None else expected_line},"
+        f"column={'*' if expected_column is None else expected_column},"
+        f"end_column{'*' if expected_end_column is None else expected_end_column})"
+    )
+
+    if match is not None:
+        with pytest.raises(ConfigValidationError, match=match):
+            raise ConfigValidationError(errors_matching_location)
+
+
+@dataclass
+class ExpectedErrorInfo:
+    expected_line: Optional[int] = None
+    expected_column: Optional[int] = None
+    expected_end_column: Optional[int] = None
+    expected_filename: str = "test.ert"
+    filename: str = "test.ert"
+    other_files: Dict[str, Union[str, FileDetail]] = None
+    match: Optional[str] = None
+
+
+def assert_multiple_errors_from_config_with(
+    contents: str,
+    expected_errors: List[ExpectedErrorInfo],
+    write_after: Callable[[TextIO], None] = None,
+):
+    for info in expected_errors:
+        assert_that_config_leads_to_error(
+            config_file_contents=contents,
+            expected_line=info.expected_line,
+            expected_column=info.expected_column,
+            expected_end_column=info.expected_end_column,
+            expected_filename=info.expected_filename,
+            filename=info.filename,
+            other_files=info.other_files,
+            match=info.match,
+            write_after=write_after,
+        )
+
+
+# "-4 -5" is parsed as the final arg, intended behavior?...
+@pytest.mark.usefixtures("use_tmpdir")
+def test_that_disallowed_argument_is_located():
+    assert_that_config_leads_to_error(
+        config_file_contents="""
+QUEUE_OPTION DOCAL MAX_RUNNING 4
+        """,
+        expected_line=2,
+        expected_column=14,
+        expected_end_column=19,
+        match="argument .* must be one of",
+    )
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+def test_that_multiple_keyword_specific_tokens_are_located():
+    contents = """
+QUEUE_OPTION DOCAL MAX_RUNNING 4
+STOP_LONG_RUNNING flase
+NUM_REALIZATIONS not_int
+ENKF_ALPHA not_float
+RUN_TEMPLATE dsajldkald/sdjkahsjka/wqehwqhdsa
+JOB_SCRIPT dnsjklajdlksaljd/dhs7sh/qhwhe
+JOB_SCRIPT non_executable_file
+NUM_REALIZATIONS 1 2 3 4 5
+NUM_REALIZATIONS
+        """
+
+    expected_errors = [
+        ExpectedErrorInfo(
+            expected_line=2,
+            expected_column=14,
+            expected_end_column=19,
+            match="argument .* must be one of",
+        ),
+        ExpectedErrorInfo(
+            expected_line=3,
+            expected_column=19,
+            expected_end_column=24,
```

Comment:
Use `pytest.parametrize` to have list of multiple test data. Then you can inline `assert_that_config_leads_to_error`